src/main/java/io/divolte/server/config/MappingConfiguration.java

```diff
@@ -0,0 +1,55 @@
+package io.divolte.server.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Objects;
+import java.util.Optional;
+
+@ParametersAreNonnullByDefault
+public class MappingConfiguration {
+    private static final String DEFAULT_DISCARD_CORRUPTED = "false";
+    private static final String DEFAULT_DISCARD_DUPLICATES = "false";
+
+    public final Optional<String> schemaFile;
+    public final Optional<String> mappingScriptFile;
+
+    public final ImmutableSet<String> sources;
+    public final ImmutableSet<String> sinks;
+
+    public final boolean discardCorrupted;
+    public final boolean discardDuplicates;
+
+    @JsonCreator
+    MappingConfiguration(final Optional<String> schemaFile,
+                         final Optional<String> mappingScriptFile,
+                         @JsonProperty(required = true)
```

Comment:
The `reference.conf` doesn't supply anything at all for sources/mappings/sinks because of the way merging works: defaults are handled in code. So this annotation means that users have to supply sources/sinks when they declare a mapping. (Without it, Jackson will happily supply `null`.)

`@NotEmpty` is interesting. Although indeed such a mapping is pointless, I can imagine that it might be useful when setting up and configuring a server to have a mapping temporarily configured to have no sources or sinks. (It might also simplify template-based configuration.)